### ISSUE TITLE
PageRevision - Cloning File bug

### ISSUE THREAD

**Issue by SeanPeterson** [ohmediaorg/file-bundle#12]
## Issue
Homepage banner always showing GMS logo when editing
I've updated the banner image for the homepage but when you go into to edit anything, it always says the file name for the banner image is "GMS-Insurance.jpg". If you make an edit and save the new content, the image gets changed to the GMS image it references. 

## Cause
When the file is being cloned, its internals are in a `lazyObjectState` 
![Screen Shot 2024-11-25 at 3 55 45 PM](https://github.com/user-attachments/assets/56e90437-b108-47ea-8330-a7dcff133098)

So as when `$this->id = null;` is deleted, the contents are not yet loaded. When it does go to load the contents, it ends up fetching the first entity from the table, which is "qcm-logo" `id = 3".

![Screen Shot 2024-11-25 at 3 37 25 PM](https://github.com/user-attachments/assets/3a40d763-a9f4-463c-8f07-96c02e5dc844)

## Re-create
You can see the bug when trying to edit the homepage on prod. It shows that logo image in the banner input
https://runqcm.ca/admin/page/revision/149/content

## Solutions?

This lazy-loading is a new concept to me. I'm wondering if we can add a `unsetId` method that we then call after the clone has occurred in the `PageContentImage`?

OR

We can also explicitly set the fields in that clone function, example
```
$this->image = $this->image;
$this->path = $this->path;
```
A bit silly, but that did work in my testing.

**Comment by SeanPeterson** [comment]
@justin-oh FYI

**Comment by justin-oh** [comment]
@SeanPeterson I just found this https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/cookbook/implementing-wakeup-or-clone.html.

Looks like we will need to update all `__clone` methods on entities to match this pattern.

**Comment by SeanPeterson** [comment]
> @SeanPeterson I just found this https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/cookbook/implementing-wakeup-or-clone.html.
> 
> Looks like we will need to update all `__clone` methods on entities to match this pattern.
> 
> I don't know much about those proxy classes either.

Need to refresh the brain cells. But this might be a pattern we can use to ensure things are correctly loaded in these clone functions

```
public function __clone()
    {
        if ($this->id) {

            if ($this instanceof Proxy) {
                $this->__load();
            }

            $this->id = null;
            $this->cloned = true;
            // don't want cloned items to appear in the File browser
            $this->browser = false;
            $this->folder = null;

            $this->resizes = new ArrayCollection();
        }
    }
```

**Comment by justin-oh** [comment]
@SeanPeterson Unsure if the Proxy check is necessary because it doesn't specifically say to do it in that documentation.

Also totally unsure if it would ever work because we are inside of the literal entity class checking `$this`. Proxies are confusing.

**Comment by SeanPeterson** [comment]
> @SeanPeterson Unsure if the Proxy check is necessary because it doesn't specifically say to do it in that documentation.
> 
> Also totally unsure if it would ever work because we are inside of the literal entity class checking `$this`. Proxies are confusing.

So `__wakeup` is never called, I confirmed this by setting a breakpoint. I don't think it's necessary to implement this. I think that documentation is kinda just outlining what not to do (*I need more research though), make sure the ID is set, don't throw or return stupid shit. 



Also the load does work. Here's before

![Screen Shot 2024-11-26 at 9 55 06 AM](https://github.com/user-attachments/assets/ada18c8f-45ac-4f37-9684-27a6836a07da)

Then after

![Screen Shot 2024-11-26 at 9 54 37 AM](https://github.com/user-attachments/assets/8565d421-1b53-4d86-b476-e99aa1699755)



But I'll keep researching into this before making any sort of PR.

**Comment by SeanPeterson** [comment]
https://stackoverflow.com/questions/16667049/why-the-need-for-safe-wakeup-and-clone-methods-in-doctrine-2

"
Because Doctrine 2 unserializes and clones entities when it needs to hydrate them, those methods will get called. But when Doctrine 2 does this, the entities will not have any data set (not even the identifier(s)). The data will be set afterwards.

So when you implement them in such a way that your logic is only performed when the entity does have the identifier(s), you are sure that your logic is only performed when you unserialize or clone the entity, not when Doctrine 2 does it.
"

TL;DR - We'll want to update our `__clone` methods to match that pattern to ensure our custom code does not get called/used by doctrine. Doubly true if we're hydrating within the clone function I presume.
